fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/ValueIndexScrubbingToolsMissing.java

```diff
@@ -0,0 +1,177 @@
+import com.apple.foundationdb.record.PipelineOperation;
+import com.apple.foundationdb.record.RecordCursor;
+import com.apple.foundationdb.record.RecordCursorResult;
+import com.apple.foundationdb.record.ScanProperties;
+import com.apple.foundationdb.record.TupleRange;
+import com.apple.foundationdb.record.logging.KeyValueLogMessage;
+import com.apple.foundationdb.record.logging.LogMessageKeys;
+import com.apple.foundationdb.record.metadata.Index;
+import com.apple.foundationdb.record.metadata.RecordType;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecord;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStore;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStoreBase;
+import com.apple.foundationdb.record.provider.foundationdb.FDBStoreTimer;
+import com.apple.foundationdb.record.provider.foundationdb.FDBStoredRecord;
+import com.apple.foundationdb.record.provider.foundationdb.IndexMaintainer;
+import com.apple.foundationdb.record.provider.foundationdb.IndexScrubbingTools;
+import com.apple.foundationdb.record.query.plan.RecordQueryPlanner;
+import com.apple.foundationdb.record.query.plan.synthetic.SyntheticRecordFromStoredRecordPlan;
+import com.apple.foundationdb.record.query.plan.synthetic.SyntheticRecordPlanner;
+import com.apple.foundationdb.tuple.Tuple;
+import com.google.protobuf.Message;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+
+/**
+ * Index Scrubbing Toolbox for a Value index maintainer. Scrub missing value index entries.
+ */
+public class ValueIndexScrubbingToolsMissing implements IndexScrubbingTools<FDBStoredRecord<Message>> {
+    private Collection<RecordType> recordTypes = null;
+    private Index index;
+    private boolean allowRepair;
+    private boolean isSynthetic;
+
+    @Override
+    public void presetCommonParams(Index index, boolean allowRepair, boolean isSynthetic, Collection<RecordType> types) {
+        this.recordTypes = types;
+        this.index = index;
+        this.allowRepair = allowRepair;
+        this.isSynthetic = isSynthetic;
+    }
+
+    @Override
+    public String getName() {
+        return "Scrub missing value index entries";
+    }
+
+    @Override
+    public RecordCursor<FDBStoredRecord<Message>> getIterator(final TupleRange tupleRange, final FDBRecordStore store, int limit) {
+        final IsolationLevel isolationLevel = IsolationLevel.SNAPSHOT;
+        final ExecuteProperties.Builder executeProperties = ExecuteProperties.newBuilder()
+                .setIsolationLevel(isolationLevel)
+                .setReturnedRowLimit(limit);
+
+        final ScanProperties scanProperties = new ScanProperties(executeProperties.build(), false);
+        return store.scanRecords(tupleRange, null, scanProperties);
```

Comment:
Will this work for synthetic records?

Reply:
I do not know. What are synthetic records?